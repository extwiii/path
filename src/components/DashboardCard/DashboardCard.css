.Dashboard-cards {
  display: flex;
  flex-direction: column;
  background-color: white;
  height: 190px;
  width: 100%;
  margin: 2px;
  padding: 5px;
}

.Dashboard-cards__title {
  flex-grow: 1 1 10%;
}

.Dashboard-cards__description {
  flex: 1 1 70%;
  display: flex;
  align-items: center;
}

.Dashboard-cards__cta {
  flex: 1 1 20%;
  display: flex;
  flex-direction: row-reverse;
  margin: 5px;
}

@media (min-width: 900px) {
  .Dashboard-cards {
    width: calc(50% - 5px);
  }
}
