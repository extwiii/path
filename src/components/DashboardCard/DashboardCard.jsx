import React from 'react';
import propTypes from 'prop-types';
import './DashboardCard.css';

const DashboardCard = ({ title, description, onClick }) => (
  <div className="Dashboard-cards">
    <div className="Dashboard-cards__title">
      <h4>{title}</h4>
    </div>
    <div className="Dashboard-cards__description">
      <p>{description}</p>
    </div>
    <div className="Dashboard-cards__cta">
      <button type="button" onClick={() => onClick(title)}>
        Get Started
      </button>
    </div>
  </div>
);

DashboardCard.propTypes = {
  title: propTypes.string.isRequired,
  description: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
};

export default DashboardCard;
