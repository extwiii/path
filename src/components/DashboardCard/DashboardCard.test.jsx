import React from 'react';
import { shallow } from 'enzyme';

import DashboardCard from './DashboardCard';

describe('<DashboardCard />', () => {
  it('renders without crash', () => {
    const wrapper = shallow(
      <DashboardCard key={1234} title="My Title" description="My Description" onClick={() => {}} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
