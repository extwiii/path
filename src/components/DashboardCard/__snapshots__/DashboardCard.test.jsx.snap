// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardCard /> renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashboardCard
    description="My Description"
    onClick={[Function]}
    title="My Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Dashboard-cards__title"
        >
          <h4>
            My Title
          </h4>
        </div>,
        <div
          className="Dashboard-cards__description"
        >
          <p>
            My Description
          </p>
        </div>,
        <div
          className="Dashboard-cards__cta"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Get Started
          </button>
        </div>,
      ],
      "className": "Dashboard-cards",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            My Title
          </h4>,
          "className": "Dashboard-cards__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Title",
          },
          "ref": null,
          "rendered": "My Title",
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            My Description
          </p>,
          "className": "Dashboard-cards__description",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Description",
          },
          "ref": null,
          "rendered": "My Description",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
            type="button"
          >
            Get Started
          </button>,
          "className": "Dashboard-cards__cta",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get Started",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Get Started",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Dashboard-cards__title"
          >
            <h4>
              My Title
            </h4>
          </div>,
          <div
            className="Dashboard-cards__description"
          >
            <p>
              My Description
            </p>
          </div>,
          <div
            className="Dashboard-cards__cta"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              Get Started
            </button>
          </div>,
        ],
        "className": "Dashboard-cards",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              My Title
            </h4>,
            "className": "Dashboard-cards__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My Title",
            },
            "ref": null,
            "rendered": "My Title",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              My Description
            </p>,
            "className": "Dashboard-cards__description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My Description",
            },
            "ref": null,
            "rendered": "My Description",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
              type="button"
            >
              Get Started
            </button>,
            "className": "Dashboard-cards__cta",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Get Started",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Get Started",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
